### PR TITLE
Describe the limitation of INDEX function

### LINKED ISSUE #978: Index formula with null 2nd arg should still work
### Description

In google sheets and excel, the index formula is able to default the 2nd argument to 1 if it is empty. Hyperformula will throw a #VALUE error. 

We should support leaving that 2nd argument blank, or update the documentation to make that limitation more clear. 

```
test("Match function on computed dates", () => {
  const engine = HyperFormula.buildFromArray(
    [
      ["1", "2", "3"],
      ["A", "B", "C"],
      ["1", "=INDEX(A2:C2, 1, MATCH(A3, A1:C1, 0))"],
      ["2", "=INDEX(A2:C2,, MATCH(A4, A1:C1, 0))"],
    ],
    {
      dateFormats: ["D/M/YY"],
      licenseKey: "gpl-v3",
    }
  );
  expect(engine.getCellValue({ sheet: 0, row: 2, col: 1 })).toEqual("A"); // Passes
  expect(engine.getCellValue({ sheet: 0, row: 3, col: 1 })).toEqual("B"); // Fails with #VALUE
});

```

### PR DESCRIPTION
**Pull request by sequba**
### Context

Our implementation of index function requires the 2 first parameters. Only the last one is optional. This behaviour is different from the other spreadsheet software so must be documented.

### How has this been tested?
build pass, unit tests pass

### Types of changes

- [x] Documentation change

### Related issue(s):
1. Fixes #978 
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [ ] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

**Review by kirszenbaum — CHANGES_REQUESTED** [review]

File: docs/guide/known-limitations.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ you can't compare the arguments in a formula like this:
 * CHISQ.INV, CHISQ.INV.RT, CHISQ.DIST.RT, CHIDIST, CHIINV and CHISQ.DIST (CHISQ.DIST in CDF mode): Running time grows linearly with the value of the second parameter, degrees_of_freedom (slow for values>1e7).
 * GAMMA.DIST, GAMMA.INV, GAMMADIST, GAMMAINV (GAMMA.DIST and GAMMADIST in CDF mode): Running time grows linearly with the value of the second parameter, alpha (slow for values>1e7). 
 * For certain inputs, the RATE function might have no solutions, or have multiple solutions. Our implementation uses an iterative algorithm (Newton's method) to find an approximation for one of the solutions to within `1e-7`. If the approximation is not found after 50 iterations, the RATE function returns the `#NUM!` error.
+* The two first parameters of the INDEX function are required. Only the last one can be omitted (it defaults to '1' in such a case).
```
[the remark this answers is not in the document]
Comment (sequba) [reply docs/guide/known-limitations.md:36]: You're right. I updated the description in `built-in-functions` page:
> Returns the content of a cell, specified by row and column number. The column parameter is optional and defaults to 1.

But this function works differently from [Excel](https://support.microsoft.com/en-us/office/index-function-a5dcf0dd-996d-40a4-a822-b56b061328bd) so it should be mentioned in `know-limitations`. The differences are:
1. Only column parameter is optional in our implementation.
2. When column parameter is not provided INDEX return a cell value from column 1; not the whole row.

[the remark this answers is not in the document]
Comment (sequba) [reply docs/guide/known-limitations.md:36]: @kirszenbaum Please feel free to describe these differences in `known-limitations` the way you think is the most appropriate.

File: docs/guide/built-in-functions.md [hunk 2]
```diff
@@ -224,19 +224,19 @@ Total number of functions: **{{ $page.functionsCount }}**
...
+| CHOOSE | Uses an index to return a value from a list of up to 30 values.                                                        | CHOOSE(Index; Value1; ...; Value30)                 |
+| COLUMN | Returns column number of a given reference or formula reference if argument not provided.                              | COLUMNS([Reference])                                |
+| COLUMNS | Returns the number of columns in the given reference.                                                                  | COLUMNS(Array)                                      |
+| FORMULATEXT | Returns a formula in a given cell as a string.                                                                         | FORMULATEXT(Reference)                              |
+| HLOOKUP | Searches horizontally with reference to adjacent cells to the bottom.                                                  | HLOOKUP(Search_Criterion; Array; Index; Sort_Order) |
+| INDEX | Returns the content of a cell, specified by row and column number. The column parameter is optional and defaults to 1. | INDEX(Range; Row[; Column])                         |
```
Comment (kirszenbaum) [inline docs/guide/built-in-functions.md:234]:
```suggestion
| INDEX | Returns the contents of a cell specified by row and column number. The column number is optional and defaults to 1. | INDEX(Range; Row [; Column])                         |
```

- It's content**s**, not content
- The comma shouldn't be there (changes the meaning)
- IMO either "the `Column` parameter" or "the column number"

File: docs/guide/known-limitations.md [hunk 3]
```diff
@@ -33,6 +33,7 @@ you can't compare the arguments in a formula like this:
 * CHISQ.INV, CHISQ.INV.RT, CHISQ.DIST.RT, CHIDIST, CHIINV and CHISQ.DIST (CHISQ.DIST in CDF mode): Running time grows linearly with the value of the second parameter, degrees_of_freedom (slow for values>1e7).
 * GAMMA.DIST, GAMMA.INV, GAMMADIST, GAMMAINV (GAMMA.DIST and GAMMADIST in CDF mode): Running time grows linearly with the value of the second parameter, alpha (slow for values>1e7). 
 * For certain inputs, the RATE function might have no solutions, or have multiple solutions. Our implementation uses an iterative algorithm (Newton's method) to find an approximation for one of the solutions to within `1e-7`. If the approximation is not found after 50 iterations, the RATE function returns the `#NUM!` error.
+* INDEX function always returns a single cell. It does not support returning whole rows/columns of the source range. The two first parameters of the INDEX function are required. Only the `Column` parameter can be omitted (it defaults to '1' in such a case).
```
Comment (kirszenbaum) [inline docs/guide/known-limitations.md:36]:
```suggestion
* The INDEX function doesn't support returning whole rows or columns of the source range – it always returns the contents of a single cell.
```

**Review by kirszenbaum — CHANGES_REQUESTED** [review]

File: docs/guide/known-limitations.md (hunk 3 above)
Comment (sequba) [reply docs/guide/known-limitations.md]: Suggestion applied

Comment (sequba) [reply docs/guide/known-limitations.md]: Thanks a lot!

File: docs/guide/built-in-functions.md (hunk 2 above)
Comment (sequba) [reply docs/guide/built-in-functions.md]: Thanks a lot!

**Review by kirszenbaum — APPROVED** [review]

**Pull request merged**